### PR TITLE
Custom Coin Field Select Component

### PR DESCRIPTION
**Pull request by youngkidwarrior**

### CODE REVIEW

**Comment by youngkidwarrior** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/442?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#442** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/442?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#441** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/441?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#439](https://github.com/0xsend/sendapp/pull/439) <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/439?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#440** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/440?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `dev`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @youngkidwarrior and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/app/data/coins.tsx
Comment (0xBigBoss) [inline packages/app/data/coins.tsx:1]: these have been reborn under data. the JSX made it tricky to share this in other packages.

Comment (youngkidwarrior) [reply packages/app/data/coins.tsx:1]: Noticed that and totally for it, didnt mean to sneak this back in

File: packages/ui/src/components/FormFields/CoinField.tsx
Comment (0xBigBoss) [inline packages/ui/src/components/FormFields/CoinField.tsx:1]: I'd move this to the app package. it has too many deps on on app files.

Comment (0xBigBoss) [reply packages/ui/src/components/FormFields/CoinField.tsx:1]: it's also breaking the build.
Screenshot 2024-06-06 at 09 40 19

Comment (youngkidwarrior) [reply packages/ui/src/components/FormFields/CoinField.tsx:1]: Honestly, all these field components should probably live in the app package. They are all very stateful

**Comment by 0xBigBoss** [comment]
lint error https://github.com/0xsend/sendapp/actions/runs/9409022949/job/25918204252?pr=442#step:4:35

File: packages/app/components/FormFields/CoinField.tsx [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+    address: sendAccount?.address,
+    token: coin.token === 'eth' ? undefined : coin.token,
+    query: { enabled: !!sendAccount },
+    chainId: baseMainnet.id,
+  })
+
+  return (
+    <Select.Item index={index} key={coin.token} value={coin.token} bc="transparent" f={1} w="100%">
+      <XStack gap={'$2'} $gtLg={{ gap: '$3.5' }} ai={'center'} jc={'space-between'}>
+        <IconX />
```
Comment (0xBigBoss) [inline packages/app/components/FormFields/CoinField.tsx:171]: should be the coin icon right?

**Review by 0xBigBoss — APPROVED** [review]
should try to get this positioning down and also show some sort of indication of which one is currently selected.

Screenshot 2024-06-06 at 19 07 14

**Comment by youngkidwarrior** [comment]
We are switching designs to this

https://www.figma.com/design/d9CZyOeYO42w4yOpeJbHES/Send-Designs-1.0?node-id=4872-11997&t=FJNfI0Fx7n4uvCOw-4

**Comment by youngkidwarrior** [comment]
### Merge activity

* **Jun 9, 1:17 AM EDT**: @youngkidwarrior started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/0xsend/sendapp/442).

**Pull request merged**